tilemap.c

```diff
@@ -29,7 +29,7 @@ tile_map_new(int size, size_t item_size, TileMapItemFreeFunc item_free_func)
     self->size = size;
     self->item_size = item_size;
     self->item_free_func = item_free_func;
-    const int map_size = 2*self->size*2*self->size;
+    const float map_size = 2*self->size*2*self->size;
```

Comment:
.. I hope I'm misunderstanding something here:

* Line 32 said `const int map_size = 2*self->size*2*self->size;` I'm not sure how the compiler refactors that, but this in combination with the stated error seems to imply self->size=40960, meaning 6,710,886,400 is the value assigned to this.
* Line 33 multiplies this by self->item_size - which seems as if it must be a pointer, therefore either 4 or 8 bytes. So it requests to malloc either 26,843,545,600 bytes or 53,687,091,200 bytes
* ie. either 25GB or 50GB.
* If this is true, why is it happening?

* Line 65 just confirms my hypothesis about the value of self->item_size
* Not sure what the line 71 report is about; there isn't any multiplying happening there.

Putting aside that allocation issue.. Isn't `size_t` the naturally appropriate type for passing to `malloc` (cf. `man malloc`)? IMO both `map_size` and `i` should be of this type.